apps/meteor/app/livechat/server/api/v1/agent.ts

```diff
@@ -58,3 +60,49 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'livechat/agent.status',
+	{ authRequired: true, validateParams: isPOSTLivechatAgentStatusProps },
+	{
+		async post() {
+			const { status } = this.bodyParams;
+			let { agentId } = this.bodyParams;
+			if (!agentId) {
+				agentId = this.userId;
+			}
+
+			const agent = await Users.findOneAgentById(agentId, {
+				projection: {
+					status: 1,
+					statusLivechat: 1,
+				},
+			});
+			if (!agent) {
+				return API.v1.failure('Agent not found');
+			}
+
+			const newStatus = status || (agent.statusLivechat === 'available' ? 'not-available' : 'available');
+			if (newStatus === agent.statusLivechat) {
+				return API.v1.success();
+			}
+
+			if (agentId !== this.userId) {
+				if (!hasPermission(this.userId, 'manage-livechat-agents')) {
+					return API.v1.unauthorized();
+				}
+				Livechat.setUserStatusLivechat(agentId, newStatus);
+
+				return API.v1.success();
+			}
+
+			if (!Livechat.allowAgentChangeServiceStatus(newStatus, agentId)) {
```

Comment:
Should this check be b4 the admin check? Should admins be allowed to bypass business hours? 👀

Reply:
Yeah, you're right. Based on the current logic, admins/managers are allowed to bypass business hours